app/models/user.rb

```diff
@@ -15,6 +17,7 @@ class User < ApplicationRecord
   has_many :sessions, dependent: :destroy
 
   scope :active, -> { where(active: true) }
+  scope :active_and_banned, -> { where(active: false).and(where(banned: true)).or(where(active: true)) }
```

Comment:
I think you could simplify this and get the same result if you go with:
```suggestion
  scope :active_and_banned, -> { where.not(active: false, banned: false) }
```

The logic:
| Active | Banned | (!A and B) or A | Result |
|:-:|:-:|:-:|:-:|
| T | T | (F and T) or T | T |
| T | F | (F and F) or T | T |
| F | T | (T and T) or F | T |
| F | F | (T and F) or F | F |